File: GooglePay/src/main/java/com/braintreepayments/api/GooglePayPaymentAuthResult.java
package com.braintreepayments.api;

import androidx.annotation.Nullable;

import com.google.android.gms.wallet.PaymentData;

/**
 * Result returned from the callback used to instantiate {@link GooglePayLauncher} that should be
 * passed to {@link GooglePayClient#tokenize(GooglePayPaymentAuthResult, GooglePayTokenizeCallback)}
 */
public class GooglePayPaymentAuthResult {

    private final PaymentData paymentData;
    private final Exception error;

    GooglePayPaymentAuthResult(@Nullable PaymentData paymentData, @Nullable Exception error) {
        this.paymentData = paymentData;
        this.error = error;
    }

    PaymentData getPaymentData() {
        return paymentData;
    }

    Exception getError() {
        return error;
    }
}

File: GooglePay/src/main/java/com/braintreepayments/api/googlepay/GooglePayPaymentAuthResult.kt
package com.braintreepayments.api.googlepay

import com.google.android.gms.wallet.PaymentData

/**
 * Result returned from the callback used to instantiate [GooglePayLauncher] that should be
 * passed to [GooglePayClient.tokenize]
 */
class GooglePayPaymentAuthResult internal constructor(
    internal val paymentData: PaymentData?,
    internal val error: Exception?
)
